### DISCUSSION TITLE
Can dynamic() imports fallback to a <Suspense/> boundary up the tree still? Or only using the new loading() parameter

### DISCUSSION THREAD

**Discussion by michael-mcgreal** [vercel/next.js#44412] [Help] [answered]
Previous next docs said that when a dynamic import suspends, it will fallback to the nearest suspense boundary up the tree.

Now, there is a loading parameter in the dynamic() interface to define the fallback component as an argument – however, defining the fallback as a loading argument always wraps the imported module as a direct parent. II cannot find a way to make it fallback to <Suspense> boundaries higher up the tree.

Does anyone know if dynamic() still allows falling back to <Suspense> boundaries higher up the tree? (instead of providing a loading() argument)

**Comment by icyJoseph** [comment, marked as answer]
I am not sure what you mean. `next/dynamic` is a sort of enhanced version of React.lazy. And both are facades or wrappers, for `import()`. It's not always the case that one would want to lazily/dynamically, load a component, but also a library. 

React.lazy's behaviour is to suspend, and it requires a Suspense wrapper. next/dynamic went ahead and tried to implement all of the above, library loading, lazy client bundle, with or without SSR, a custom loading placeholder, etc. The package itself has evolved quite a bit, but there's a few gotchas with React 18.

In React 18 Server Components, the loading is awaited, so it might not be behaving at all as you expect it. 

If we follow the docs:

> There are two ways you can implement lazy loading in Next.js:
> 
> Recommended: Using React.lazy() with [Suspense](https://beta.reactjs.org/apis/react/Suspense). Please see the [React documentation](https://beta.reactjs.org/apis/react/lazy) for examples.
> Using [Dynamic Imports (next/dynamic)](https://beta.nextjs.org/docs/optimizing/lazy-loading#nextdynamic) if you're migrating from pages to app.
> 

And now a couple of clarification questions, are you on app, or pages? Couldn't you just use React.lazy instead?

**Reply by michael-mcgreal** [reply]
Thanks @icyJoseph! 

Using Reacy.lazy() directly achieved the behavior I was seeking, where suspense would fallback to a suspense boundary _a few nodes up the tree_ (whereas next/dynamic seems to implement a suspense boundary in the component, preventing falling back further – which is great in many cases!).

To answer your other questions (for future reader context), I was using the pages directory in next13.
